### PR TITLE
[wip] Review Module 10 - REST

### PR DESCRIPTION
**Pull request by millidavids**
Uses https://github.com/millidavids/rest-client-ex/blob/master/api-test.php to test. Keep getting a 401. Signatures and HMAC don't seem to work. I could use some guidance.

Library I am using: https://github.com/delatbabel/apisecurity

@bkuhl @jlaswell 

You can see where I am going with this. Using a RSA signature library, you encode the request using the parameters and attempt to decode the request parameters on arrival (I believe this is how amazon handles time sensitive urls from s3, s3 generated a time sensitive url with a signature, upon decoding if the timestamp is overdue, render a 401). On a success, render content to execute the post or put or delete; of a failure render a 401. You can see that I know HOW it is supposed to work, it just seems like I am struggling with a library at this point.

### CODE REVIEW

**Comment by millidavids** [comment]
You can see where I am going with this. Using a RSA signature library, you encode the request using the parameters and attempt to decode the request parameters on arrival (I believe this is how amazon handles time sensitive urls from s3). On a success, render content to execute the post or put or delete; of a failure render a 401. You can see that I know HOW it is supposed to work, it just seems like I am struggling with a library at this point.

File: app/Http/Controllers/CompanyController.php [hunk 1]
```diff
@@ -4,58 +4,82 @@
...
+        $server->setPublicKey(base_path().'/public_key');
+        try
+        {
+            $server->verifyHMAC($params);
 
-        return redirect('/company');
+        } catch (SignatureException $e) {
+            return false;
+        }
+        return true;
```
Comment (bkuhl) [inline app/Http/Controllers/CompanyController.php:83]: This is really good functionality to put in a [middleware](https://laravel.com/docs/master/middleware#introduction).  Nice to see you got it all in place!

Comment (bkuhl) [inline app/Http/Controllers/CompanyController.php:83]: Also.. just doing some knowledge sharing here... multiple forms of auth support has been added to L5 - see https://github.com/laravel/laravel/blob/master/config/auth.php#L38.  I haven't used this yet... maybe @jlaswell has?

Comment (bkuhl) [inline app/Http/Controllers/CompanyController.php:83]: It doesn't look like that signs requests though... I'd have to dig into it to find out

**Comment by bkuhl** [comment]
:+1: Looks good.. you've demonstrated understanding which is the point.  Time to move to the next thing!

**Pull request closed without merging**